src/components/BooksQuery.tsx

```diff
@@ -1,45 +1,44 @@
-import axios from "axios";
 import { useState } from "react";
 import Card from "@mui/material/Card";
+import { v4 as uuidv4 } from "uuid";
 import "./books.css";
 import { CardContent, Typography } from "@mui/material";
-type book = {
-  id: number;
-  title: string;
-  author: string;
-  year: string;
-};
+import BookObject from "../types/book";
+import AllBooks from "../types/allBooks";
+import ResponseObject from "../types/respBooks";
 const BooksQuery = () => {
   const [book, setBooks] = useState("");
-  const [list, setList] = useState([]);
+  const [list, setList] = useState<AllBooks>({ books: [] });
   const submit = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
     e.preventDefault();
     if (!book) {
       alert("Please enter a book name");
     } else {
-      getData();
+      getData().then((data) => {
```

Comment:
use async/await

Reply:
getData is function already used async/await !

Reply:
but submit is not !!!